Don't delete a Docker volume if the volume's driver differs

rhansen opened the pull request:
Currently, if an existing volume's driver does not match the specified driver, `states.dockerng.volume_present()` deletes and re-creates the volume.  To avoid surprising the user with data loss, this PR introduces a new `force` parameter—defaulting to `False`—that controls this behavior.

These commits add `versionadded` and `versionchanged` documentation that assume these commits will make it into the 2015.8.6 release.  Please let me know if I should change them.

terminalmage commented on salt/states/dockerng.py:
```diff
@@ -2100,7 +2105,16 @@ def volume_present(name, driver=None, driver_opts=None):
-        Option for tha volume driver
+        Options for the volume driver
+
+    force
```
Can you change this line to ``force : False`` to indicate the default value?

rhansen committed:
typo fixes

rhansen committed:
add 'force' to replace (or not) volumes with driver mismatch

Add a new 'force' option that will cause volume_present() to delete
and re-create a volume if the existing volume's driver doesn't match
the specified driver.  Set the default to True to preserve existing
behavior.

rhansen committed:
change default for volume_present()'s force parameter to False

This is safer, as it avoids surprising the user with permanent data
loss.

rhansen commented:
I pushed a new version of the commits.  Changes:

  * rebased onto latest develop
  * changed docstring documentation to mention default value

ticosax commented:
@rhansen please, can you make this PR against 2015.8 branch ?
It will be later integrated into develop by saltstack's staff.

Otherwise :+1: on my side

cachedout commented:
@ticosax I'll just mark this as needing a backport and we'll take care of getting into 2015.8. Thanks!

rhansen commented:
Thank you all for reviewing and merging.

@ticosax: I interpreted the [SaltStack Git Policy documentation](https://docs.saltstack.com/en/latest/topics/development/git/index.html#new-code-entry) as indicating that pull requests should always be based on the `develop` branch, even if the intention is for the commits to eventually land on an existing release branch.  Am I misunderstanding that page?

rhansen commented:
Hmm, I just found the ["Which Salt Branch?"](https://docs.saltstack.com/en/latest/topics/development/contributing.html#which-salt-branch) page, which suggests that I should have based my commits on the 2015.8 branch.  I also just noticed that the Policy documentation I referenced earlier says, "Code contributors to SaltStack projects DO NOT NEED TO READ THIS DOCUMENT."  Sorry about that!

rallytime commented:
@rhansen Perhaps that Git Policy doc is a little out of date. The Which Salt Branch is the one you should follow. Apologies for the confusion there. I've back-ported this to 2015.8 in #30941. Thank you very much for your contribution!
